rockcraft/services/image.py

```diff
@@ -65,25 +67,28 @@ def _create_image_info(self) -> ImageInfo:
         image_dir = self._work_dir / "images"
         bundle_dir = self._work_dir / "bundles"
 
-        if len(self._build_plan) != 1:
+        build_plan = self._services.get("build_plan").plan()
+
+        if len(build_plan) != 1:
             raise errors.MultipleBuildsError
 
-        build_for = self._build_plan[0].build_for
-        project = cast(models.Project, self._project)
-        if project.base == "bare":
+        build_for = build_plan[0].build_for
+        project = self._services.get("project").get()
+        base = cast(str, project.effective_base)
```

Comment:
this must be the project's actual `base`, and *not* the "build-base if build-base else base"